File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
const halvesAreAlike = s => {
    let vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
    let left = right = 0
    for(let i = 0; i<s.length; i++){
        if(vowels.includes(s[i])){
            i < s.length/2 ? left++ : right++
        }
    }
    return left == right
};

File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.rs
impl Solution {
    pub fn halves_are_alike(string: String) -> bool {
        let vowels = "aeiouAEIOU".to_string();
        let mut left = 0;
        let mut right = 0;
        for (index, char) in string.chars().enumerate(){
            if vowels.contains(char) {
                if index < string.len()/2 { left+=1 }
                else{ right += 1}
            }
        }
        left == right
    }
}